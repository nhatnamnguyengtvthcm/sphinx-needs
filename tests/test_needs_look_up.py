import json
from pathlib import Path

import pytest


@pytest.mark.parametrize(
    "test_app", [{"buildername": "needs_lut", "srcdir": "doc_test/doc_needs_builder"}], indirect=True
)
def test_doc_needs_id_builder(test_app):
    app = test_app
    app.build()

    needs_json = Path(app.outdir, "needs_lut.json")
    with open(needs_json) as needs_file:
        needs_file_content = needs_file.read()

    needs_list = json.loads(needs_file_content)
    assert needs_list["TC_NEG_001"]
