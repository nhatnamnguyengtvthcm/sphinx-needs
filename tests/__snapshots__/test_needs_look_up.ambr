# serializer version: 1
# name: test_doc_needs_lut_builder[test_app0]
  dict({
    'current_version': '1.0',
    'project': 'Python',
    'versions': dict({
      '1.0': dict({
        'filters': dict({
        }),
        'filters_amount': 0,
        'needs': dict({
          'TC_001': dict({
            'description': '',
            'docname': 'index',
            'id': 'TC_001',
          }),
          'TC_NEG_001': dict({
            'description': '',
            'docname': 'index',
            'id': 'TC_NEG_001',
          }),
          'US_63252': dict({
            'description': '',
            'docname': 'index',
            'id': 'US_63252',
          }),
        }),
        'needs_amount': 3,
      }),
    }),
  })
# ---
